pkg/gowork/system.go

```diff
@@ -0,0 +1,54 @@
+package gowork
+
+import (
+	"fmt"
+	"io/fs"
+	"os"
+	"path"
+	"path/filepath"
+)
+
+const rootDir = "/"
+
+var (
+	systemFS                 = os.DirFS(rootDir) //nolint:gochecknoglobals
+	_        operatingsystem = RealSystem{}
+)
+
+// RealSystem is the real, live OS implementation.
+type RealSystem struct{}
+
+func (s RealSystem) Get(name string) string {
+	return os.Getenv(name)
+}
+
+func (s RealSystem) Open(name string) (fs.File, error) {
+	f, err := systemFS.Open(name)
+	if err != nil {
+		return nil, fmt.Errorf("%w: %v", ErrBug, err)
+	}
+	return f, nil
+}
+
+func (s RealSystem) Cwd() (string, error) {
+	dir, err := os.Getwd()
+	if err != nil {
+		return "", fmt.Errorf("%w: %v", ErrBug, err)
+	}
+	p, err := filepath.Rel(rootDir, dir)
+	if err != nil {
+		return "", fmt.Errorf("%w: %v", ErrBug, err)
+	}
+	return p, nil
+}
+
+func (s RealSystem) Abs(filepath string) string {
+	return path.Clean(path.Join(rootDir, filepath))
+}
+
+// OS represents a virtual operating system.
+type operatingsystem interface {
+	FileSystem
+	Environment
+	Abs(filepath string) string
```

Comment:
Do we need a new interface here, or something that can take a `FileSystem` and `path string` and return the absolute path?

Reply:
In particular, I'm worried about hoisting what was an implementation in the `modscope` CLI into a general-purpose module without good reason for expanding the interface.  Can we keep the `operatingsystem` abstraction in `modscope/cli`?

Reply:
Well, this `operatingsystem` interface is private. It isn't used anywhere. The `modscope/cli` have different `OS` interface (compatible OFC).

I use it just to simplify the ensure assignment on line 15:

```golang
var _ operatingsystem = RealSystem{}
```

Without it, I would need to type:

```golang
var (
  _ FileSystem  = RealSystem{}
  _ Environment = RealSystem{}
)
```

and that doesn't ensure the `Abs(filepath string) string` func.

OFC I could live without those assures...